src/subscription/__tests__/subscribe-test.js

```diff
@@ -545,22 +545,22 @@ describe('Subscribe', () => {
     });
   });
 
-  it('invalid query should result in error', async () => {
-    const invalidAST = parse(`
+  it('unknown field should result in closed subscription', async () => {
+    const ast = parse(`
       subscription {
-        invalidField
+        unknownField
       }
     `);
 
-    expect(() => {
-      subscribe(
-        emailSchema,
-        invalidAST,
-      );
-    }).to.throw('This subscription is not defined by the schema.');
+    const pubsub = new EventEmitter();
+
+    const { subscription } = createSubscription(pubsub, emailSchema, ast);
+
+    const payload = await subscription.next();
+    expect(payload).to.deep.equal({ done: true, value: undefined });
```

Comment:
In typical validate-then-operate usage, most definitely. Attempting to use an undefined field is a validation error.

However we need to handle these corner cases nonetheless. In the case that validation was thwarted in some way. Perhaps a doc was validated before being persisted, and then a breaking change was made to a schema which caused the doc to become invalid - but future runs of it have skipped validation due to being persisted.

This is the sort of thing that's rare, but needs definition.